v2/fb-xcom-api-validator/tests/matchers/to-be-listed-asset.ts

```diff
@@ -0,0 +1,55 @@
+import { AssetDefinition, NotFoundError } from '../../src/client/generated';
+import { MatcherFunction, SyncExpectationResult } from 'expect';
+import { SecureClient as Client } from '../../src/client/SecureClient';
+
+declare global {
+  // eslint-disable-next-line @typescript-eslint/no-namespace
+  namespace jest {
+    interface AsymmetricMatchers {
+      toBeListedAsset(): void;
+    }
+    interface Matchers<R> {
+      toBeListedAsset(): R;
+    }
+  }
+}
+
+export const toBeListedAsset: MatcherFunction<[assetId: unknown]> = async function (
```

Comment:
Matchers should never fail themselves and never have side effects. This matcher sends HTTP requests that might fail. In that case you are forcing the user to debug the matcher.

My idea was to have a general matcher that can run any predicate. I checked now and there is such a matcher in `jest-extended` package. Sorry I didn't think of it earlier. Here, take a look:
https://jest-extended.jestcommunity.dev/docs/matchers/tosatisfy

Reply:
Since this matcher doesn't support async functions, I would need to call the function outside of the matcher and validate the result, which is no different than just doing `expect(functionResult).toBe(true)`
So I'll stick with basic jest here and remove my matcher.